magenta-lib/src/main/scala/magenta/tasks/ASGTasks.scala

```diff
@@ -162,16 +162,14 @@ case class ResumeAlarmNotifications(pkg: DeploymentPackage, stage: Stage, stack:
 class ASGResetException(message: String, throwable: Throwable) extends Throwable(message, throwable)
 
 trait ASGTask extends Task {
+  def asgName: String
   def region: Region
-  def pkg: DeploymentPackage
-  def stage: Stage
-  def stack: Stack
 
   def execute(asg: AutoScalingGroup, resources: DeploymentResources, stopFlag: => Boolean, asgClient: AutoScalingClient)
 
   override def execute(resources: DeploymentResources, stopFlag: => Boolean) {
     ASG.withAsgClient(keyRing, region, resources) { asgClient =>
-      val group = ASG.groupForAppAndStage(pkg, stage, stack, asgClient, resources.reporter)
```

Comment:
This is the main change described by:

> Previously, Riff-Raff would search for the relevant ASG (by tags) as part of every individual task (e.g. WaitForStabilization) in a deployment action (e.g. deploy).

> Each individual task can then [now] simply ask for the latest state of the specific ASG it cares about, rather than searching through all ASGs in the account.

Reply:
Does this mean the `riff-raff.yaml` would need to list the ASGs to work on explicitly?

Reply:
We still find the ASG using tags (so there are no changes to the `riff-raff.yaml`), it just happens once per `deploy` action ([here](https://github.com/guardian/riff-raff/pull/631/files#diff-c63b6a3e501bff7096061d2ceaf6514397e592cd3440d40a5dc5cf0b96d72f9dR72)) instead of once per task. Happy to chat about this via a call if it's hard to understand (it took me a while to get my head around this code)!